_1327/documents/templates/documents_edit.html

```diff
@@ -403,6 +403,26 @@ <h4 class="modal-title" id="deleteAutosaveHeader">{% trans "Do you really want t
 		// add datepicker to date inputs
 		$("input[name*='date']:not([readonly='True'])").datepicker( $.datepicker.regional["{{ LANGUAGE_CODE }}"] );
 
+		function getCurrentTime() {
+			const currentDate = new Date();
+			const currentHour = currentDate.getHours().toString();
+			let currentMinute = currentDate.getMinutes().toString();
+			if(currentMinute.length == 1) currentMinute = "0" + currentMinute;
+
+			return currentHour + ":" + currentMinute;
+		}
+
+		function insertText(textToInsert, markdownEditor) {
+			const markdownText = markdownEditor[0].value;
+			const cursorPosition = markdownEditor[0].selectionStart;
```

Comment:
Should the insertion position always be the cursor position? One could argue that the `|start|(...)` text should always be placed at the beginning, while the `|end|(...)` should always be inserted at the end, even when the user is still changing some stuff in the middle.
Then again, I'm not sure how deterministic this feels for users. Especially, I'm not sure how inserting text into text fields and changing its selection affects the scrolling position. If the text field is rather small and users hit the end button repeatedly without realizing new text gets inserted somewhere they can't see it, that would certainly be worse than the current implementation.

Reply:
both should be placed at the cursor position. there might be some introduction before the beginning time and similarly there can be text after the ending mark.